internal/kafka/internal/services/clusters.go

```diff
@@ -503,14 +506,14 @@ func (c clusterService) FindAllClusters(criteria FindClusterCriteria) ([]*api.Cl
 	// They are mostly the same as the library we use (xid) does take the generation timestamp into consideration,
 	// However, it only down to the level of seconds. This means that if a few records are created at almost the same time,
 	// the order is not guaranteed. So use the `created_at` column will provider better consistency.
-	if err := dbConn.Where(clusterDetails).Order("created_at asc").Scan(&cluster).Error; err != nil {
+	if err := dbConn.Where(clusterDetails).Order("created_at asc").Scan(&clusters).Error; err != nil {
```

Comment:
Side comment:

For this method we should document that attributes in `FindClusterCriteria` whose value is the zero-value of their corresponding type are not included in the search criteria